semeio/jobs/design_kw/design_kw.py

```diff
@@ -79,15 +79,50 @@ def is_comment(line):
 
 
 def extract_key_value(parameters):
-    # Extract all key, value pairs from parameters file
-    # If key already exists, raise Exception
+    """Parses a list of strings, looking for key-value pairs pr. line
+    separated by whitespace, into a dictionary.
 
+    Spaces in either key or in value are not supported.
+
+    Args:
+        parameters (list of str)
+
+    Returns:
+        dict, with the keys and values parsed.
+
+    Raises:
+        ValueError, with error messages and all unparseable lines.
+    """
     res = {}
+    unparsable_lines = []
+    errors = []
     for line in parameters:
-        key, value = line.split()
+        try:
+            key, value = line.split()
+        except ValueError as e_msg:
+            if "not enough values to unpack" in str(
+                e_msg
+            ) or "need more than 1 value to unpack" in str(e_msg):
+                # (py2 and py3 differ in emitted error message)
+                errors += ["No value found in line {}".format(line)]
+            elif "too many values to unpack" in str(e_msg):
+                errors += ["Spaces in values are not supported"]
+            else:
+                errors += [str(e_msg)]
```

Comment:
I am not sure parsing the error message is the best approach here, it would for example be possible to use `len(line.split())` in the error message and say for example `Expected 2, found: {} in line: {}).format(len(line.split()), line)` or something of that nature.

Reply:
Rewritten to catch the error before  the `ValueError` occurs. I have kept different errors for `len() == 1` and `len() > 2`, I  think it  benefits  users.

Reply:
This change also introduced  ignoring empty lines. The code in the master branch will error on empty lines. Not sure if that is an error, or if it is something we should ignore. Comments are already ignored, so it can make sense to  ignore empty lines also.